src/main/java/fr/jmmc/oiexplorer/core/gui/chart/RulerOverlay.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package fr.jmmc.oiexplorer.core.gui.chart;
+
+import java.awt.BasicStroke;
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import java.awt.geom.Point2D;
+import java.awt.geom.Rectangle2D;
+import org.jfree.chart.ChartMouseEvent;
+import org.jfree.chart.ChartPanel;
+import org.jfree.chart.axis.ValueAxis;
+import org.jfree.chart.panel.AbstractOverlay;
+import org.jfree.chart.panel.Overlay;
+import org.jfree.chart.plot.PlotRenderingInfo;
+import org.jfree.chart.plot.XYPlot;
+import org.jfree.chart.ui.RectangleEdge;
+
+/**
+ * This class handles the overlay of the ruler
+ *
+ * @author martin
+ */
+public class RulerOverlay extends AbstractOverlay implements Overlay, EnhancedChartMouseListener {
+
+    private boolean isRulerActivated = false;
+    private int rulerX1, rulerY1, rulerX2, rulerY2;
+    private double[] origin, end;
+    private int clickCounter = 0;
```

Comment:
clickCounter seems not very useful ?
Maybe related to isRulerActivated => rulerState = new enum (DISABLED, EDITING, DONE) ?